docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,239 @@
+        baseBranch: <base_branch_name>
+        targetBranch: <target_branch_name>
+        secret:
+          namespace: argocd
+          name: git-creds
+    
+status:
+...
+  # imageUpdate stores the history of image updates that have been
+  # conducted, and contains information about which image was last 
+  # updated, its version and update timestamp, in order to respect 
+  # application rollbacks and avoid repeat update attempts
+  imageUpdates:
+  - path: quay.io/some/image
+    alias:  someImage
+    updatedAt: <update_timestamp>
+    updatedTo: v1.1.0
+     
+```
+
+There is a need to introduce state by tracking the status of image Updates in the application CR. This is mainly important to honor application rollbacks, avoid repeated image updates in case of failure, as well as to support future use cases such as enabling image-updater to create pull requests against the source repositories. The above detailed `.status.imageUpdates` field could be extended to store metadata regarding pull request creation that would serve to avoid duplicate/excessive number of pull requests being opened, for example.
+
+There could be a few different options in terms of placement for this block of information:
+
+1. `.status.imageUpdates` - As shown above, we could introduce a new status field at the highest level that could store a slice of `imageUpdates` containing information about the last version updated to and timestamp of last update etc. This would be a non breaking change and easy to implement/maintain/extend
+2. `.status.summary.images` - Currently this field exists as a slice of strings that just contains the full qualified paths of all images associated to an application. This filed could be repurposed by changing it from `[]string` to `[]imageUpdate` to store the additional information above. This seems like a naturally better place for it, however this would be a breaking change (and at present  would also break Image Updater as it uses this field to get the list of potentially updateable images)
+3. `.status.summary.imageUpdates` - This field could instead be introduced within the applicaiton summary with the understanding that `status.summary.images` will be deprecated and eventually removed, since it will be rendered obsolete by this new field. 
+
+## Open Questions [optional]
+
+Based on this proposal, the Image Updater controller would need to write to the application status fields in order to track the state of image updates for various images. This would mean we could potentially have 2 controllers writing to the application resource. Since the introduction of server side apply in Argo CD, it is not uncommon to have multiple field managers controlling different fields in a resource, however it may be the case that this does not extend to the status fields. In such an event, should just the responsibility of updating the application status field for image updates be shifted to the application-controller? 
```

Comment:
See my comment above where I listed the problem in having it as a separate controller.